src/renderer/components/modal/tx/TxModal.tsx

```diff
@@ -56,41 +43,24 @@ export const TxModal: React.FC<Props> = (props): JSX.Element => {
   )
 
   const renderExtra = useMemo(() => <Styled.SubContentRow>{extra}</Styled.SubContentRow>, [extra])
-
-  const renderTxButton = useCallback(
-    (txHash) => (
-      <Styled.ViewTxButton onClick={() => onViewTxClick(txHash)} key={txHash}>
-        {intl.formatMessage({ id: 'common.viewTransaction' })}
-      </Styled.ViewTxButton>
-    ),
-    [intl, onViewTxClick]
+  const renderExtraResult = useMemo(
+    () => (extraResult ? <Styled.ExtraResultContainer>{extraResult}</Styled.ExtraResultContainer> : <></>),
+    [extraResult]
   )
 
-  const renderResultDetails = useMemo(
+  const renderResult = useMemo(
     () => (
-      <Styled.ResultDetailsContainer>
-        <Styled.BtnCopyWrapper>
-          <Styled.ViewButton
-            disabled={RD.isInitial(txRD) || RD.isPending(txRD)}
-            color="success"
-            onClick={RD.isSuccess(txRD) ? onFinish : onClose}>
-            {intl.formatMessage({ id: RD.isFailure(txRD) ? 'common.cancel' : 'common.finish' })}
-          </Styled.ViewButton>
-
-          {FP.pipe(
-            oTxHash,
-            // render view tx button if
-            // 1. txHash property has been set
-            O.map(renderTxButton),
-            // or
-            // 2. txRD is successfull
-            O.alt(() => FP.pipe(RD.toOption(txRD), O.map(renderTxButton))),
-            O.getOrElse(() => <></>)
-          )}
-        </Styled.BtnCopyWrapper>
-      </Styled.ResultDetailsContainer>
+      <Styled.ResultContainer>
+        <Styled.ResultButton
+          disabled={RD.isInitial(txRD) || RD.isPending(txRD)}
+          color="success"
+          onClick={RD.isSuccess(txRD) ? onFinish : onClose}>
+          {intl.formatMessage({ id: RD.isFailure(txRD) ? 'common.cancel' : 'common.finish' })}
+        </Styled.ResultButton>
+        {renderExtraResult}
+      </Styled.ResultContainer>
     ),
-    [intl, oTxHash, onClose, onFinish, renderTxButton, txRD]
+    [intl, onClose, onFinish, renderExtraResult, txRD]
```

Comment:
just IMHO it's better to keep things more explicit without any conditions at the "view"-layer. such approach provides clearer code for understanding and maintaining
```tsx
const renderResult = useMemo(() => {
    const resultButtonProps = FP.pipe(
      txRD,
      RD.fold(
        () => ({
          disabled: true,
          onClick: onClose,
          children: intl.formatMessage({ id: 'common.finish' })
        }),
        () => ({
          disabled: true,
          onClick: onClose,
          children: intl.formatMessage({ id: 'common.finish' })
        }),
        () => ({
          onClick: onClose,
          children: intl.formatMessage({ id: 'common.cancel' })
        }),
        () => ({
          onClick: onFinish,
          children: intl.formatMessage({ id: 'common.finish' })
        })
      )
    )

    return (
      <Styled.ResultContainer>
        <Styled.ResultButton color="success" {...resultButtonProps} />
        {renderExtraResult}
      </Styled.ResultContainer>
    )
  }, [intl, onClose, onFinish, renderExtraResult, txRD])
```